src/gui/src/dbDescriptors.cpp

```diff
@@ -2579,4 +2607,171 @@ void DbModuleDescriptor::getModules(odb::dbModule* module, SelectionSet& objects
   }
 }
 
+//////////////////////////////////////////////////
+
+DbTechViaDescriptor::DbTechViaDescriptor(odb::dbDatabase* db) :
+  db_(db)
+{
+}
+
+std::string DbTechViaDescriptor::getName(std::any object) const
+{
+  auto* via = std::any_cast<odb::dbTechVia*>(object);
+  return via->getName();
+}
+
+std::string DbTechViaDescriptor::getTypeName() const
+{
+  return "Tech Via";
+}
+
+bool DbTechViaDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void DbTechViaDescriptor::highlight(std::any object,
+                                    Painter& painter,
+                                    void* additional_data) const
+{
+  return;
+}
+
+Descriptor::Properties DbTechViaDescriptor::getProperties(std::any object) const
+{
+  auto* via = std::any_cast<odb::dbTechVia*>(object);
+  auto* gui = Gui::get();
+
+  Properties props;
+
+  odb::dbTechLayer* cut_layer = nullptr;
+  for (auto* box : via->getBoxes()) {
+    if (box->getTechLayer()->getType() == odb::dbTechLayerType::CUT) {
+      cut_layer = box->getTechLayer();
+      break;
+    }
+  }
+  props.push_back({"Bottom layer", gui->makeSelected(via->getBottomLayer())});
```

Comment:
It would be nice to give the boxes on each layer

Reply:
Added in 6c381f0